pqarrow/arrowutils/sort.go

```diff
@@ -0,0 +1,108 @@
+package arrowutils
+
+import (
+	"bytes"
+	"fmt"
+	"sort"
+
+	"github.com/apache/arrow/go/v14/arrow"
+	"github.com/apache/arrow/go/v14/arrow/array"
+	"github.com/apache/arrow/go/v14/arrow/memory"
+	"golang.org/x/exp/constraints"
+
+	"github.com/polarsignals/frostdb/pqarrow/builder"
+)
+
+// SortRecord sorts the given record's rows by the given column. Currently only supports int64, string and binary columns.
+func SortRecord(r arrow.Record, cols []int) ([]int, error) {
```

Comment:
Can we also add `direction []int` which will behave similar  `bytes.Compare` does

- `-1` for ascending
- `0` no direction
- `1` for descending

This will simply sorting , in the less function you can return `bytes.Compare(...)==direction`